oak/common/app_config.bzl

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright 2020 The Project Oak Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Rule for serializing an Oak application configuration."""
+
+def serialized_config(name, textproto, modules):
+    """Serializes an Oak application configuration in a binary file.
+
+    Args:
+        name: Name of the build rule.
+        textproto: Textproto file with application configuration.
+        modules: A dictionary with module names as keys and module paths as values.
+    """
+    srcs = [textproto] + modules.values()
+    output = name + ".bin"
+    module_list = ",".join([n + ":$(location " + path + ")" for (n, path) in modules.items()])
+    cmd = "$(location //oak/common:app_config_serializer)" + \
```

Comment:
Is the trailing `\` needed? I don't see it in other similar files: https://github.com/google/asylo/blob/ebdff40ca061a6f733812cc50a88e403759c6a14/asylo/bazel/asylo.bzl#L265-L269

Reply:
Strangely enough, but it only works, if I use it as a parameter to a rule.
For example, this will compile:
```python
native.genrule(
        name = name,
        srcs = srcs,
        outs = [output],
        cmd = "$(location //oak/common:app_config_serializer)" + 
          " --textproto=$(location {})".format(textproto) + 
          " --modules={}".format(module_list) + 
          " --output_file=$@",
        tools = ["//oak/common:app_config_serializer"],
    )
```
And this will not:
```python
cmd = "$(location //oak/common:app_config_serializer)" + 
          " --textproto=$(location {})".format(textproto) + 
          " --modules={}".format(module_list) + 
          " --output_file=$@"
    native.genrule(
        name = name,
        srcs = srcs,
        outs = [output],
        cmd = cmd,
        tools = ["//oak/common:app_config_serializer"],
    )
```
```
ERROR: /opt/my-project/oak/common/app_config.bzl:30:62: syntax error at 'newline': expected expression
ERROR: /opt/my-project/oak/common/app_config.bzl:40:5: Trailing comma is allowed only in parenthesized tuples.
ERROR: /opt/my-project/oak/common/app_config.bzl:30:62: contains syntax error(s)
ERROR: /opt/my-project/oak/common/app_config.bzl:31:61: contains syntax error(s)
ERROR: /opt/my-project/oak/common/app_config.bzl:37:9: name 'outs' is not defined
ERROR: /opt/my-project/oak/common/app_config.bzl:32:47: cannot assign to '+$error$'
ERROR: /opt/my-project/oak/common/app_config.bzl:39:9: name 'tools' is not defined (did you mean 'bool'?)
ERROR: /opt/my-project/oak/common/app_config.bzl:40:6: contains syntax error(s)
```